grails-app/services/au/org/ala/collectory/CollectoryAuthService.groovy

```diff
@@ -1,41 +1,126 @@
 package au.org.ala.collectory
 
+import au.org.ala.web.AuthService
 import au.org.ala.ws.security.client.AlaAuthClient
+import org.grails.web.servlet.mvc.GrailsWebRequest
+import org.pac4j.core.credentials.AnonymousCredentials
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.context.request.RequestContextHolder
+import javax.servlet.http.HttpServletRequest
 
-
-
+/**
+ * Require TokenInterceptor to authenticate token and inject authorisation profile into the request for M2M call
+ */
 class CollectoryAuthService{
     static transactional = false
     def grailsApplication
-    def authService
     def providerGroupService
+    AuthService authService
 
     @Autowired(required = false)
     AlaAuthClient alaAuthClient
 
-
-    def username() {
+    /**
+     * @return full name of the user, or 'not available' if not authenticated
+     */
+    String username() {
+        //both work
         def username = authService.getDisplayName()
+//        def request = getRequest()
+//        def profiles = request?.profileManager?.context?.request?.profiles
+//        def username = (profiles && !profiles.isEmpty()) ? profiles[0]?.attributes?.get('username')?.toString() : null
+        //short the statement
+        return username ?: 'not available'
+    }
 
-        return (username) ? username : 'not available'
+    String userEmail() {
+        String email = authService.getEmail()
+        return email
     }
 
+
     def isAdmin() {
-        return !grailsApplication.config.security.oidc.enabled.toBoolean() || authService.userInRole(grailsApplication.config.ROLE_ADMIN as String)
+        def request = getRequest()
+        return request?.isUserInRole(grailsApplication.config.ROLE_ADMIN as String)
+    }
+
+    def isAuthenticated() {
```

Comment:
Remove `isAuthenticated()` if not in use.

Reply:
removed